src/org/zaproxy/zap/view/widgets/WritableFileChooser.java

```diff
@@ -24,6 +24,11 @@ public WritableFileChooser(File currentDirectory) {
 	@Override
 	public void approveSelection() {
 		File selectedFile = getSelectedFile();
+		
+		if (selectedFile.length() > selectedFile.getUsableSpace()) {
```

Comment:
I think doing it here was my suggestion. If that's not workable then the next best bet for this specific case is to add a try/catch for ioexception (is it always ioexception when full?) To ReportLastScan, showing a dialog that suggests the disk may be full and the file empty or incomplete... @thc202 what do you think?